google/cloud/testing_util/validate_metadata.cc

```diff
@@ -89,7 +89,13 @@ RoutingHeaders ExtractMDFromHeader(std::string header) {
  */
 MATCHER_P(MatchesGlob, glob, "matches the glob: \"" + glob + "\"") {
```

Comment:
To give a concrete example... If we did this with spanner's `BatchCreateSessions`:

https://github.com/googleapis/googleapis/blob/dd546bf83f7aa6dd24e17d3e83d9a397a6dd680c/google/spanner/v1/spanner.proto#L89

Then `FromHttpRule` would return `{{"database", "projects/*/instances/*/databases/*"}}`

https://github.com/googleapis/google-cloud-cpp/blob/c91050bbc2153ac3e5bcf43818e7bc2930d0648b/google/cloud/testing_util/validate_metadata.cc#L221-L224

But we could have that `FromHttpRule` accept the `method` and the `request`, the way `FromRoutingRule` does.

**Disclaimer:** I am not sure this approach is worth doing though. It could be a time trap.